extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/querykit/scan/ScanQueryFrameProcessor.java

```diff
@@ -223,27 +224,32 @@ protected ReturnOrAwait<Long> runWithInputChannel(
     }
   }
 
-  private void populateFrameWriterAndFlushIfNeeded() throws IOException
+  private void populateFrameWriterAndFlushIfNeeded()
   {
     createFrameWriterIfNeeded();
 
-    while (!cursor.isDone()) {
-      if (!frameWriter.addSelection()) {
-        if (frameWriter.getNumRows() > 0) {
-          final long numRowsWritten = flushFrameWriter();
+    try {
+      while (!cursor.isDone()) {
+        if (!frameWriter.addSelection()) {
+          if (frameWriter.getNumRows() > 0) {
+            final long numRowsWritten = flushFrameWriter();
 
-          if (runningCountForLimit != null) {
-            runningCountForLimit.addAndGet(numRowsWritten);
-          }
+            if (runningCountForLimit != null) {
+              runningCountForLimit.addAndGet(numRowsWritten);
+            }
 
-          return;
-        } else {
-          throw new FrameRowTooLargeException(currentAllocatorCapacity);
+            return;
+          } else {
+            throw new FrameRowTooLargeException(currentAllocatorCapacity);
+          }
         }
-      }
 
-      cursor.advance();
-      partitionBoostVirtualColumn.setValue(partitionBoostVirtualColumn.getValue() + 1);
+        cursor.advance();
+        partitionBoostVirtualColumn.setValue(partitionBoostVirtualColumn.getValue() + 1);
+      }
+    }
+    catch (Exception e) {
```

Comment:
Woudn't FrameRowTooLargeException be caught here?

Reply:
Thanks for catching this. I have reduced the surface of the try..catch block so as to not obstruct any other exceptions